components/dashboard/src/components/PageWithSubMenu.tsx

```diff
@@ -20,30 +23,64 @@ export interface PageWithSubMenuProps {
-                    <ul className="flex flex-col text tracking-wide text-gray-500 pt-4 lg:pt-0 w-52 space-y-2">
+                    <ul
+                        className={classNames(
+                            // Handle flipping between row and column layout
+                            "flex flex-row md:flex-col items-center",
+                            "w-full md:w-52 overflow-auto md:overflow-visible",
+                            "pt-4 pb-4 md:pb-0",
+                            "space-x-2 md:space-x-0 md:space-y-2",
+                            "tracking-wide text-gray-500",
+                        )}
+                    >
                         {p.subMenu.map((e) => {
-                            let classes = "flex block py-2 px-4 rounded-md";
-                            if (e.link.some((l) => l === location.pathname)) {
-                                classes += " bg-gray-300 text-gray-800 dark:bg-gray-800 dark:text-gray-50";
-                            } else {
-                                classes += " hover:bg-gray-100 dark:hover:bg-gray-800";
-                            }
-                            return (
-                                <Link to={e.link[0]} key={e.title}>
-                                    <li className={classes}>{e.title}</li>
-                                </Link>
-                            );
+                            return <SubmenuItem title={e.title} link={e.link} key={e.title} />;
                         })}
                     </ul>
                 </div>
-                <div className="ml-16 lg:ml-32 w-full pt-1 mb-40">{p.children}</div>
+                <div className="md:ml-16 lg:ml-32 w-full pt-1 mb-40">
+                    <Separator className="md:hidden" />
+                    <div className="pt-4 md:pt-0">{p.children}</div>
+                </div>
             </div>
         </div>
     );
 }
+
+type SubmenuItemProps = {
+    title: string;
+    link: string[];
+};
+
+const SubmenuItem: FC<SubmenuItemProps> = ({ title, link }) => {
+    const location = useLocation();
+    const itemRef = useRef<HTMLLIElement>(null);
+
+    // TODO: can remove this once we use sub-routes and don't re-render the whole page for each sub-route
+    useEffect(() => {
+        if (itemRef.current && link.some((l) => l === location.pathname)) {
+            itemRef.current.scrollIntoView({ behavior: "auto", block: "nearest", inline: "start" });
+        }
+    }, [link, location.pathname]);
+
+    let classes = "flex block py-2 px-4 rounded-md whitespace-nowrap max-w-52";
+
+    if (link.some((l) => l === location.pathname)) {
+        classes += " bg-gray-300 text-gray-800 dark:bg-gray-800 dark:text-gray-50";
+    } else {
+        classes += " hover:bg-gray-100 dark:hover:bg-gray-800";
+    }
+    return (
+        <Link to={link[0]} key={title} className="md:w-full">
+            <li ref={itemRef} className={classes}>
+                {title}
+            </li>
+        </Link>
```

Comment:
**suggestion**: Probably not the best line to comment about this, but could we not shift the menu items when an item is clicked? Noticed that clicking on an item moves it to the left side of the scrollable area, which sometimes seems a bit jarring, no?


https://user-images.githubusercontent.com/120486/227506186-68bcb395-dc8b-470b-af7e-679f09f2e797.mov

Reply:
Yah, I didn't like this either, tried to explain it a bit in the PR description, but it's buried in there:

> Currently navigation on the submenu results in a full route re-render, which resets the scroll position of the menu. This PR shifts the active item into visibility, but it isn't able to maintain scroll position due to the re-render. We can address this in a followup by using nested routes w/ react-router, and only re-render the portions of the page that change as much as possible.

Another thing to consider is if we did a dropdown for nav instead of this scrollbar or a tab-bar, it would avoid that issue.